src/redis_disk.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#pragma once
+
+#include <string>
+#include "redis_db.h"
+#include "redis_metadata.h"
+
+namespace Redis {
+
+class Disk : public Database {
+ public:
+  explicit Disk(Engine::Storage *storage, const std::string &ns): Database(storage, ns) {
+    this->option_.include_memtabtles = true;
+    this->option_.include_files = true;
+  }
+  rocksdb::Status GetApproximateSizes(const Metadata &metadata, const Slice &ns_key,
+                                      rocksdb::ColumnFamilyHandle *column_family,
+                                      uint64_t *key_size, Slice subkeyleft = Slice(),
+                                      Slice subkeyright = Slice());
+  rocksdb::Status GetStringSize(const Slice &user_key, uint64_t *key_size);
```

Comment:
Should we put these functions in each type of files (Like `redis_string.h` and `redis_string.cpp`)?

Reply:
@caipengbo Do you mean that each type should implement its own function (e.g. `GetApproximateKeySize`) ? Otherwise, I think, it's not the best idea to spread functions of `Disk` class among different files (one function per file).

Reply:
I think `GetApproximateKeySize` should be a virtual function and every data type should rewrite it

Reply:
> I think `GetApproximateKeySize` should be a virtual function and every data type should rewrite it

Using virtual functions is also a good approach. May I ask  using virtual functions still require switch to create different objects? If so,that looks like it's about the same as my current implementation. Maybe my c++ is not proficient enough, do you have a better solution?

Reply:
Indeed, we need to create different objects. We added an additional command to calculate the size of different types, and your implementation is also a good method.